File: app/src/main/java/com/fazemeright/chatbotmetcs622/database/messages/MessageDao.java
package com.fazemeright.chatbotmetcs622.database.messages;

import androidx.room.Dao;
import androidx.room.FtsOptions;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.fazemeright.chatbotmetcs622.database.BaseDao;

import java.util.List;

@Dao
public interface MessageDao extends BaseDao<Message> {

  @Query("SELECT * from my_messages_table WHERE mid = :key")
  Message get(long key);

  @Query("DELETE FROM my_messages_table")
  void clear();

  @Query("SELECT * FROM my_messages_table ORDER BY timestamp DESC")
  List<Message> getAllMessages();

  @Query("SELECT * FROM my_messages_table WHERE chatRoomId = :chatRoomId ORDER BY timestamp DESC")
  List<Message> getAllMessagesFromChatRoom(long chatRoomId);

  @Query("DELETE from my_messages_table WHERE chatRoomId = :chatRoomId")
  void clearChatRoomMessages(long chatRoomId);

  @Query(
      "SELECT * FROM my_messages_table WHERE chatRoomId = :chatRoomId ORDER BY timestamp DESC LIMIT 1")
  Message getLatestMessage(long chatRoomId);

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  void insertAllMessages(List<Message> order);
}

File: app/src/main/java/com/fazemeright/chatbotmetcs622/database/message/MessageDao.kt
package com.fazemeright.chatbotmetcs622.database.message

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.fazemeright.chatbotmetcs622.database.BaseDao

@Dao
interface MessageDao : BaseDao<Message?> {
    @Query("SELECT * from my_messages_table WHERE mid = :key")
    operator fun get(key: Long): Message?

    @Query("DELETE FROM my_messages_table")
    fun clear()

    @get:Query("SELECT * FROM my_messages_table ORDER BY timestamp DESC")
    val allMessages: List<Message?>?

    @Query("SELECT * FROM my_messages_table WHERE chatRoomId = :chatRoomId ORDER BY timestamp DESC")
    fun getAllMessagesFromChatRoom(chatRoomId: Long): List<Message?>?

    @Query("SELECT * FROM my_messages_table WHERE chatRoomId = :chatRoomId ORDER BY timestamp DESC")
    fun getAllMessagesFromChatRoomLive(chatRoomId: Long): LiveData<List<Message?>?>?

    @Query("DELETE from my_messages_table WHERE chatRoomId = :chatRoomId")
    fun clearChatRoomMessages(chatRoomId: Long)

    @Query("SELECT * FROM my_messages_table WHERE"
            + " chatRoomId = :chatRoomId ORDER BY timestamp DESC LIMIT 1")
    fun getLatestMessage(chatRoomId: Long): Message?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllMessages(messages: List<Message?>?)
}